File: demo/vb/ApartmentTests.vb
Imports System.Threading
Imports NUnit.Framework

Namespace NUnitTestDemo

    <Apartment(ApartmentState.STA)>
    Public Class FixtureWithApartmentAttributeOnClass

        <Test>
        Public Sub TestMethodInSTAFixture()
            Assert.That(Thread.CurrentThread.GetApartmentState(), Iz.EqualTo(ApartmentState.STA))
        End Sub

    End Class

    Public Class FixtureWithApartmentAttributeOnMethod

        <Test, Apartment(ApartmentState.STA)>
        Public Sub TestMethodInSTA()
            Assert.That(Thread.CurrentThread.GetApartmentState(), Iz.EqualTo(ApartmentState.STA))
        End Sub

    End Class

End Namespace

File: demo/cpp/ApartmentTests.cpp
using namespace System;
using namespace System::Threading;
using namespace NUnit::Framework;

namespace NUnitTestDemo
{
    [Apartment(ApartmentState::STA)]
    public ref class FixtureWithApartmentAttributeOnClass
    {
	public:
        [Test]
        void TestMethodInSTAFixture()
        {
            Assert::That(Thread::CurrentThread->GetApartmentState(), Is::EqualTo(ApartmentState::STA));
        }
	};

    public ref class FixtureWithApartmentAttributeOnMethod
    {
	public:
        [Test, Apartment(ApartmentState::STA)]
        void TestMethodInSTA()
        {
            Assert::That(Thread::CurrentThread->GetApartmentState(), Is::EqualTo(ApartmentState::STA));
        }
	};
}
